examples/flutter_firebaseauth/lib/authentication/authentication_service.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+class FirebaseAuthService {
+  Future<FirebaseUser> getInitialSignInState() {
+    return FirebaseAuth.instance.onAuthStateChanged.first;
+  }
+
+  Future<FirebaseUser> sigInWithGoogle() async {
+    FirebaseAuth _instance = FirebaseAuth.instance;
+    //Signin to google
+    final GoogleSignIn _googleSignIn = GoogleSignIn();
+    final GoogleSignInAccount googleUser = await _googleSignIn.signIn();
+    final GoogleSignInAuthentication googleAuth =
+        await googleUser.authentication;
+
+    final AuthCredential credential = GoogleAuthProvider.getCredential(
+      accessToken: googleAuth.accessToken,
+      idToken: googleAuth.idToken,
+    );
+
+    //Signin to Firebase and pass on the google credential
+    final FirebaseUser user =
+        await _instance.signInWithCredential(credential).then((fbUser) {
```

Comment:
in what case would the email and displayName be null? is there any value in asserting that? If i'm not mistaken, asserts are removed from release builds.

Reply:
you are right,removed asserts. It was just a copy-paste overlook.